proxy.c

```diff
@@ -184,13 +185,21 @@ void sigterm_handler(int signal) {
     exit(0);
 }
 
+int set_reuseaddr(int sock)
```

Comment:
Could be done with fixing line 148 to optval=1 (two additional characters) instead of a whole new function (six lines of code), or am I missing something?